src/main/java/net/snowflake/ingest/streaming/internal/FileColumnProperties.java

```diff
@@ -284,6 +288,16 @@ void setMaxStrNonCollated(String maxStrNonCollated) {
     this.maxStrNonCollated = maxStrNonCollated;
   }
 
+  @JsonProperty("numberOfValues")
+  @JsonInclude(value = JsonInclude.Include.CUSTOM, valueFilter = IgnoreMinusOneFilter.class)
```

Comment:
instead of the customer IgnoreMinusOne logic why not just have  @JsonInclude(JsonInclude.Include.NON_NULL), and let numberOfValues be null by default / null for FDN ingestion?

Reply:
Switched to `Long` and use`NON_NULL`.